recipes/IWSLT22_lowresource/train.py

```diff
@@ -0,0 +1,409 @@
+        else:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                sort_key="duration"
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                sort_key="duration"
+            )
+
+        hparams["dataloader_options"]["shuffle"] = False
+        hparams["dataloader_options"]["shuffle"] = False
+    elif hparams["sorting"] == "descending":
+        # use smaller dataset to debug the model
+        if hparams["debug"]:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                key_min_value={"duration": hparams["sorting_min_duration"]},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+                reverse=True,
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                key_min_value={"duration": hparams["sorting_min_duration"]},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+                reverse=True,
+            )
+        else:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                sort_key="duration", reverse=True
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                sort_key="duration", reverse=True
+            )
+
+        hparams["dataloader_options"]["shuffle"] = False
+        hparams["dataloader_options"]["shuffle"] = False
+    elif hparams["sorting"] == "random":
+        # use smaller dataset to debug the model
+        if hparams["debug"]:
+            datasets["train"] = datasets["train"].filtered_sorted(
+                key_min_value={"duration": 3},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+            )
+            datasets["valid"] = datasets["valid"].filtered_sorted(
+                key_min_value={"duration": hparams["sorting_min_duration"]},
+                key_max_value={"duration": hparams["sorting_max_duration"]},
+                sort_key="duration",
+            )
+
+        hparams["dataloader_options"]["shuffle"] = True
+    else:
+        raise NotImplementedError(
+            "sorting must be random, ascending or descending"
+        )
+
+    return datasets, tokenizer
+
+
+if __name__ == "__main__":
+
+    # Load hyperparameters file with command-line overrides
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+
+    # If distributed_launch=True then
+    # create ddp_group with the right communication protocol
+    sb.utils.distributed.ddp_init_group(run_opts)
+
+    # Create experiment directory
+    sb.create_experiment_directory(
+        experiment_directory=hparams["output_folder"],
+        hyperparams_to_save=hparams_file,
+        overrides=overrides,
+    )
+
```

Comment:
Usually, we call the data preparation here with a proper run_on_main call (see other recipes) :-)

Reply:
@mzboito can the data preparation be moved/called here via `run_on_main` ?

Reply:
Hello @anautsch , can you please check if you have the last version of my code?
I already did this change: see exchange below with Titouan and line 391 here https://github.com/mzboito/speechbrain/blob/iwslt_speech_translation/recipes/IWSLT22_lowresource/train.py

Reply:
Hi @mzboito quick response !
Github didn't mark the file as outdates, but you are right. I have two tabs open right now, one with the change and another one without - sorry - I'll resolve this one, you are right.